lib/ansible/cli/galaxy.py

```diff
@@ -969,39 +991,111 @@ def _display_role(gr):
                 version = "(unknown version)"
             display.display("- %s, %s" % (gr.name, version))
 
-        if context.CLIARGS['role']:
-            # show the requested role, if it exists
-            name = context.CLIARGS['role']
-            gr = GalaxyRole(self.galaxy, self.api, name)
-            if gr.metadata:
-                display.display('# %s' % os.path.dirname(gr.path))
-                _display_role(gr)
+        path_found = False
+        warnings = []
+        galaxy_type = context.CLIARGS['type']
+        if context.CLIARGS['type'] == 'role':
+            if context.CLIARGS['role']:
+                # show the requested role, if it exists
+                name = context.CLIARGS['role']
+                gr = GalaxyRole(self.galaxy, self.api, name)
+                if gr.metadata:
+                    display.display('# %s' % os.path.dirname(gr.path))
+                    _display_role(gr)
```

Comment:
path_found isn't set so this particular code path fails.

Reply:
> You could probably move this for path in collections_search_paths + appending the warnings before if collection_name: so you don't need to write it for the else below too.

Brilliant! I did that and the code is now far less redundant.

Reply:
> path_found isn't set so this particular code path fails.
I see what you mean. I need to rework the role section since the collection section is gathering everything and displaying at the end rather than as encountered.

Yet another reason I need to break these up. They are becoming far too intertwined.